Source/Controllers/MainViewControllers/SPDatabaseDocument.m

```diff
@@ -2698,94 +2698,44 @@ - (IBAction)saveConnectionSheet:(id)sender
             return;
         }
 
-        // Load accessory nib each time.
-        // Note that the top-level objects aren't released automatically, but are released when the panel ends.
-        if (![NSBundle.mainBundle loadNibNamed:@"SaveSPFAccessory" owner:self topLevelObjects:nil]) {
-            NSLog(@"SaveSPFAccessory accessory dialog could not be loaded.");
-            return;
-        }
-
         // Save current session (open connection windows as SPF file)
         [panel setAllowedFileTypes:@[SPFileExtensionDefault]];
 
-        //Restore accessory view settings if possible
-        if ([spfDocData objectForKey:@"save_password"]) {
-            [saveConnectionSavePassword setState:[[spfDocData objectForKey:@"save_password"] boolValue]];
-        }
-        if ([spfDocData objectForKey:@"auto_connect"]) {
-            [saveConnectionAutoConnect setState:[[spfDocData objectForKey:@"auto_connect"] boolValue]];
-        }
-        if ([spfDocData objectForKey:@"encrypted"]) {
-            [saveConnectionEncrypt setState:[[spfDocData objectForKey:@"encrypted"] boolValue]];
-        }
-        if ([spfDocData objectForKey:@"include_session"]) {
-            [saveConnectionIncludeData setState:[[spfDocData objectForKey:@"include_session"] boolValue]];
-        }
-        if ([[spfDocData objectForKey:@"save_editor_content"] boolValue]) {
-            [saveConnectionIncludeQuery setState:[[spfDocData objectForKey:@"save_editor_content"] boolValue]];
-        }
-        else {
-            [saveConnectionIncludeQuery setState:NSOnState];
-        }
+        [self prepareSaveAccessoryViewWithPanel:panel];
 
-        [saveConnectionIncludeQuery setEnabled:([[[[customQueryInstance valueForKeyPath:@"textView"] textStorage] string] length])];
+        [self.saveConnectionIncludeQuery setEnabled:([[[[customQueryInstance valueForKeyPath:@"textView"] textStorage] string] length])];
 
         // Update accessory button states
         [self validateSaveConnectionAccessory:nil];
 
         // TODO note: it seems that one has problems with a NSSecureTextField inside an accessory view - ask HansJB
-        [[saveConnectionEncryptString cell] setControlView:saveConnectionAccessory];
-        [panel setAccessoryView:saveConnectionAccessory];
+        [[self.saveConnectionEncryptString cell] setControlView:self.saveConnectionAccessory];
+        [panel setAccessoryView:self.saveConnectionAccessory];
 
-        // Set file name
-        filename = ([[[self fileURL] path] length]) ? [self displayName] : [NSString stringWithFormat:@"%@", [self name]];
+        // Set file name to the name of the connection
+        filename = [self name];
```

Comment:
Does this work even for quick connect connections (non-favorites)?

Reply:
From what I've tried, yes. 😄